tests/test_primitives.py

```diff
@@ -0,0 +1,209 @@
+        self.assertIsInstance(self.v1 - self.v2, Vector)
+
+        # A few simple cases that should pass
+        self.assertEqual(Vector(1, 3) - Vector(0, 3), Vector(1, 0))
+        self.assertEqual(Vector(500) - Vector(400), Vector(100))
+
+        # Should only work on vectors of like sizes
+        # big - small
+        self.assertRaises(ValueError, Vector(1, 2, 3).__sub__, Vector(1, 2))
+        # small - big
+        self.assertRaises(ValueError, Vector(1, 2).__sub__, Vector(1, 2, 3))
+
+        # A few things could go horribly, horribly wrong such
+        # that the following test fails:
+        # Unfortunately it seems to be failing just because of floating
+        # point arithmetic.
+        self.assertEqual(self.v2 - self.v1 + self.v1, self.v1 - self.v1 + self.v2)
+
+
+    def test_scalar_multiplication(self):
+        """There should be some way to multiply vectors by scalars."""
+
+        self.assertEqual(1 * self.v1, self.v1 * 1)
+        self.assertEqual(0 * self.v1, self.zero_v)
+        self.assertIsInstance(5 * self.v1, Vector)
+
+    def test_dot_products(self):
+        """Tests vector dot products"""
+
+        v1 = self.v1 # The code looks so much better with these short
+        v2 = self.v2
+        zero_v = self.zero_v
+
+        # Symmetric
+        self.assertEqual(v1.dot(v2), v2.dot(v1))
+
+        # Associative with scalar multiplication
+        self.assertEqual(v1.dot(v2 * 5),
+                         v1.dot(v2) * 5)
+
+        # Norm >= 0
+        self.assertTrue(v1.dot(v1) >= 0)
+        self.assertTrue(v2.dot(v2) >= 0)
+
+        # zero is orthogonal to everything
+        self.assertEqual(v1.dot(zero_v), 0)
+        self.assertEqual(zero_v.dot(v1), 0)
+
+        # Other things that should be orthogonal are so.
+        self.assertEqual(Vector(1, 0).dot(Vector(0, 1)), 0)
+
+    def test_iteration(self):
+        """We wanted to be able to iterate over vectors"""
+
+        # Make sure you get the elements in order.
+        # Note that v1 was made by iterating over the elements of
+        # a list.
+        self.assertEqual(Vector(*[i for i in self.v1]), self.v1)
+
+    def test_indexing(self):
+        """Make sure accessing by index works"""
+        my_vector = Vector(1, 2, 3, 4, 5)
+        self.assertEqual(my_vector[0], 1)
+        self.assertEqual(my_vector[-1], 5)
+
+class PointTestCase(unittest.TestCase):
+    """Unit tests for the Point class."""
+
+    def setUp(self):
+        """Make some objects that are useful for most tests."""
+        self.a = Point(0, 0)
+        self.b = Point(1, 2)
+        self.c = Point(1, 2, 3)
+
+
+    def test_point_equal(self):
+        """Tests if comparison of points for equality works"""
+        self.assertTrue(Point(0, -9, 11.2) == Point(0, -9, 11.2))
+        self.assertFalse(Point(0, -9, 1600002) == Point(0, -8, 1600002))
+        # Different location
+        self.assertFalse(Point(1, 4) == Point(1, 4, 5))
+
+        # Drew's tests:
```

Comment:
This seems weird to me because it makes `git blame` wrong now, any way to fix that?